pkg/probe/firewall_ippool.go

```diff
@@ -0,0 +1,91 @@
+package probe
+
+import (
+	"log"
+
+	"github.com/bluecmd/fortigate_exporter/pkg/http"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type IpPool struct {
+	Name      string  `json:"name"`
+	IPTotal   int     `json:"natip_total"`
+	IPInUse   int     `json:"natip_in_use"`
+	Clients   int     `json:"clients"`
+	Available float64 `json:"available"`
+	Used      int     `json:"used"`
+	Total     int     `json:"total"`
+}
+
+type IpPoolResponse struct {
+	Results map[string]IpPool `json:"results"`
+	VDOM    string            `json:"vdom"`
+	Version string            `json:"version"`
+}
+
+func probeFirewallIpPool(c http.FortiHTTP, meta *TargetMetadata) ([]prometheus.Metric, bool) {
+	var (
+		mAvailable = prometheus.NewDesc(
+			"fortigate_ippool_available",
+			"Percentage available in ippool",
+			[]string{"vdom", "name"}, nil,
+		)
+	)
+	var (
+		mIpUsed = prometheus.NewDesc(
+			"fortigate_ippool_used_ip",
+			"Ip addresses in use in ippool",
+			[]string{"vdom", "name"}, nil,
+		)
+	)
+	var (
+		mIpTotal = prometheus.NewDesc(
+			"fortigate_ippool_total_ip",
```

Comment:
Nit pick: You use _clients (plural, which I like as well) and _ip (singular). I *personally* would use _ips but I will leave this decision on naming on you.

Reply:
changed to be plural everywhere